plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -371,6 +549,49 @@ private function build_query(): void {
 		$this->sql = "SELECT $found_rows $distinct $fields FROM $orders_table $join WHERE $where $groupby $orderby $limits";
 	}
 
+	/**
+	 * JOINs the main orders table with another table.
+	 *
+	 * @param string  $table      Table name (including prefix).
+	 * @param string  $alias      Table alias to use. Defaults to $table.
+	 * @param string  $on         ON clause. Defaults to "wc_orders.id = {$alias}.order_id".
+	 * @param boolean $alias_once If TRUE, table won't be JOIN'ed again if already JOIN'ed.
+	 * @return void
+	 * @throws \Exception When an error occurs, such as trying to re-use an alias with $alias_once = FALSE.
+	 */
+	public function join( string $table, string $alias = '', string $on = '', bool $alias_once = false ) {
+		$alias = empty( $alias ) ? $table : $alias;
+
+		if ( $this->tables['orders'] === $alias ) {
+			// translators: %s is a table name.
+			throw new \Exception( sprintf( __( '%s can not be used as a table alias in OrdersTableQuery', 'woocommerce' ), $alias ) );
+		}
+
+		if ( empty( $on ) ) {
+			if ( $this->tables['orders'] === $table ) {
+				$on = "{$this->tables['orders']}.id = {$alias}.id";
+			} else {
+				$on = "{$this->tables['orders']}.id = {$alias}.order_id";
+			}
+		}
+
+		if ( isset( $this->join[ $alias ] ) ) {
+			if ( ! $alias_once ) {
+				// translators: %s is a table name.
+				throw new \Exception( sprintf( __( 'Can not re-use table alias "%s" in OrdersTableQuery.', 'woocommerce' ), $alias ) );
+			}
+
+			return;
+		}
+
+		$sql_join  = '';
+		$sql_join .= "INNER JOIN {$table} ";
```

Comment:
should this be a `LEFT JOIN`, eventually we may have tables that are not necessary to have rows for each order.

Reply:
Got it.